crates/services/shared-sequencer/src/lib.rs

```diff
@@ -0,0 +1,280 @@
+                data.clone(),
+            )
+            .await?;
+
+        let used_gas = http_api::estimate_transaction(
+            &self.endpoints.blockchain_rest_api,
+            dummy_payload,
+        )
+        .await?;
+
+        let used_gas = used_gas.saturating_mul(2); // Add some buffer
+
+        let amount = Coin {
+            amount: self.gas_price.saturating_mul(used_gas as u128),
+            denom: self.coin_denom.clone(),
+        };
+
+        let fee = Fee::from_amount_and_gas(amount, used_gas);
+        let payload = self
+            .make_payload(timeout_height, fee, signer, account, order, topic, data)
+            .await?;
+
+        let r = self.tendermint()?.broadcast_tx_sync(payload).await?;
+        if r.code.is_err() {
+            return Err(PostBlobError { message: r.log }.into());
+        }
+        Ok(())
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    async fn make_payload<S: Signer>(
+        &self,
+        timeout_height: u32,
+        fee: Fee,
+        signer: &S,
+        account: AccountMetadata,
+        order: u64,
+        topic: [u8; 32],
+        data: Bytes,
+    ) -> anyhow::Result<Vec<u8>> {
+        let sender_account_id = self.sender_account_id(signer)?;
+
+        let msg = MsgPostBlob {
+            from: sender_account_id.to_string(),
+            order: order.to_string(),
+            topic: Bytes::from(topic.to_vec()),
+            data,
+        };
+        let any_msg = cosmrs::Any {
+            type_url: "/fuelsequencer.sequencing.v1.MsgPostBlob".to_owned(),
+            value: msg.encode_to_vec(),
+        };
+        let tx_body = tx::Body::new(vec![any_msg], "", timeout_height);
+
+        let sender_public_key = signer.public_key();
+        let signer_info =
+            SignerInfo::single_direct(Some(sender_public_key), account.sequence);
+        let auth_info = signer_info.auth_info(fee);
+        let sign_doc =
+            SignDoc::new(&tx_body, &auth_info, &self.chain_id, account.account_number)
+                .map_err(|err| anyhow!("{err:?}"))?;
+
+        let sign_doc_bytes = sign_doc
+            .clone()
+            .into_bytes()
+            .map_err(|err| anyhow!("{err:?}"))?;
+        let signature = signer.sign(&sign_doc_bytes).await?;
+
+        // Convert the signature to non-normalized form
+        let mut signature_bytes = *signature;
+        signature_bytes[32] &= 0x7f;
+
+        Ok(cosmos_sdk_proto::cosmos::tx::v1beta1::TxRaw {
+            body_bytes: sign_doc.body_bytes,
+            auth_info_bytes: sign_doc.auth_info_bytes,
+            signatures: vec![signature_bytes.to_vec()],
+        }
+        .to_bytes()?)
+    }
+
+    /// Return all blob messages in the given blob
+    pub async fn read_block_msgs(&self, height: u32) -> anyhow::Result<Vec<MsgPostBlob>> {
+        let mut result = Vec::new();
+
+        let block = self.tendermint()?.block(height).await?;
+
+        for item in block.block.data.iter().skip(1) {
```

Comment:
Why do we skip one ?

Reply:
@Dentosal Added it, so I don't know=) I will remove this function for now, since it is not used